app/src/main/java/org/simple/clinic/removeoverdueappointment/RemoveOverdueAppointmentScreen.kt

```diff
@@ -0,0 +1,137 @@
+import java.util.UUID
+import javax.inject.Inject
+
+class RemoveOverdueAppointmentScreen : BaseScreen<
+    RemoveOverdueAppointmentScreen.Key,
+    ScreenRemoveOverdueAppointmentBinding,
+    RemoveOverdueModel,
+    RemoveOverdueEvent,
+    RemoveOverdueEffect>(), RemoveOverdueUi, RemoveOverdueUiActions {
+
+  @Inject
+  lateinit var router: Router
+
+  @Inject
+  lateinit var effectHandlerFactory: RemoveOverdueEffectHandler.Factory
+
+  private val removalReasonsAdapter = ItemAdapter(
+      diffCallback = RemoveAppointmentReasonItem.DiffCallback(),
+      bindings = mapOf(
+          R.layout.contactpatient_removeappointment_reasonitem to { layoutInflater, parent ->
+            ContactpatientRemoveappointmentReasonitemBinding.inflate(layoutInflater, parent, false)
+          }
+      )
+  )
+
+  private val doneButton
+    get() = binding.removeAppointmentDone
+
+  private val toolbar
+    get() = binding.removeAppointmentToolbar
+
+  private val removalReasonsRecyclerView
+    get() = binding.removalReasonsRecyclerView
+
+  override fun defaultModel() = RemoveOverdueModel.create(screenKey.appointmentId, screenKey.patientId)
+
+  override fun bindView(layoutInflater: LayoutInflater, container: ViewGroup?) =
+      ScreenRemoveOverdueAppointmentBinding.inflate(layoutInflater, container, false)
+
+  override fun uiRenderer() = RemoveOverdueUiRenderer(this)
+
+  override fun events() = Observable.merge(
+      removeAppointmentReasonClicks(),
+      toolbarNavigationClicks(),
+      doneClicks()
+  ).compose(ReportAnalyticsEvents())
+      .cast<RemoveOverdueEvent>()
+
+  override fun createUpdate() = RemoveOverdueUpdate()
+
+  override fun createEffectHandler() = effectHandlerFactory
+      .create(this)
+      .build()
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    context.injector<Injector>().inject(this)
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    removalReasonsRecyclerView.adapter = removalReasonsAdapter
+  }
+
+  override fun renderAppointmentRemoveReasons(reasons: List<RemoveAppointmentReason>, selectedReason: RemoveAppointmentReason?) {
+    removalReasonsAdapter.submitList(RemoveAppointmentReasonItem.from(reasons, selectedReason))
+  }
+
+  override fun disableDoneButton() {
+    doneButton.isEnabled = false
+  }
+
+  override fun enableDoneButton() {
+    doneButton.isEnabled = true
+  }
+
+  override fun goBack() {
+    router.pop()
+  }
+
+  private fun removeAppointmentReasonClicks(): Observable<RemoveOverdueEvent> {
+    return removalReasonsAdapter
+        .itemEvents
+        .ofType<RemoveAppointmentReasonItem.Event.Clicked>()
+        .map { it.reason }
+        .map(::RemoveAppointmentReasonSelected)
+  }
+
+  private fun toolbarNavigationClicks(): Observable<RemoveOverdueEvent> {
+    return toolbar
+        .navigationClicks()
+        .map { CloseClicked }
+  }
```

Comment:
Yeah, I usually follow that approach but I felt like we are not capturing that flow in tests. So that's why I ended up making it part of the loop for this feature. WDYT?

Reply:
I only prefer to capture things in tests that add value. A purely navigation related functionality doesn't add too much value, but a considerable amount of maintenance burden, so I prefer not to make these part of the loop.

Reply:
Alright, let me remove that event and push it.